### PR TITLE
49 Fixed small error where iterator could go out of bounds and remove redundancy.

### PR DESCRIPTION
**Pull request by CodingAllan**
## Merge Request - GuideLine Checklist 

**Guideline** to check code before resolve WIP and approval, respectively.
As many checkboxes as possible should be ticked.

### Checks by code author:
Always to be checked:
* [x] There is at least one issue associated with the pull request.
* [x] New code adheres with the [coding guidelines](https://github.com/mknaranja/GMGPolar/wiki)
* [x] No large data files have been added to the repository. Maximum size for files should be of the order of KB not MB. In particular avoid adding of pdf, word, or other files that cannot be change-tracked correctly by git.

If functions were changed or functionality was added:
* [ ] Tests for new functionality has been added
* [ ] A local test was succesful

If new functionality was added:
* [x] There is appropriate **documentation** of your work. (use doxygen style comments)

If new third party software is used:
* [x] Did you pay attention to its license? Please remember to add it to the wiki after successful merging.

If new mathematical methods or epidemiological terms are used:
* [ ] Are new methods referenced? Did you provide further documentation?

[ ] The following questions are addressed in the documentation (if need be): 
* Developers (what did you do?, how can it be maintained?)
* For users (how to use your work?)
* For admins (how to install and configure your work?)

* For documentation: Please write or update the Readme in the current working directory!

### Checks by code reviewer(s):
* [ ] Is the code clean of development artifacts e.g., unnecessary comments, prints, ...
* [ ] The ticket goals for each associated issue are reached or problems are clearly addressed (i.e., a new issue was introduced).
* [ ] There are appropriate **unit tests** and they pass.
* [ ] The git history is clean and linearized for the merge request. All reviewers should squash commits and write a simple and meaningful commit message.
* [ ] Coverage report for new code is acceptable. 
* [ ] No large data files have been added to the repository. Maximum size for files should be of the order of KB not MB. In particular avoid adding of pdf, word, or other files that cannot be change-tracked correctly by git.

### CODE REVIEW

File: src/define_coarse_nodes.cpp [hunk 1]
```diff
@@ -124,17 +117,18 @@ void level::define_coarse_nodes_onelevel(level* coarser)
...
-    coarser->r      = std::vector<double>(nr);
+    coarser->r      = std::vector<double>(coarser->nr);
     coarser->ntheta = (int)indices_theta.size();
-    coarser->theta  = std::vector<double>(ntheta);
+    coarser->theta  = std::vector<double>(coarser->ntheta);
 
     itr_r = indices_r.begin();
-    for (int i = 0; i < nr; i++) {
+    for (int i = 0; i < coarser->nr; i++) {
         coarser->r[i] = r[*itr_r];
```
Comment (mknaranja) [inline src/define_coarse_nodes.cpp:126]: Where does this `r` come from?

File: src/define_coarse_nodes.cpp [hunk 2]
```diff
@@ -97,22 +97,15 @@ void level::define_coarse_nodes_onelevel(level* coarser)
...
-                }
+
+            if (i % 2 == 0 && j % 2 == 0) {
+                coarse_nodes_list_r[j * ntheta_int + i]     = j;
+                coarse_nodes_list_theta[j * ntheta_int + i] = i;
+
+                indices_r.insert(j);
+                indices_theta.insert(i);
             }
             else {
```
Comment (mknaranja) [inline src/define_coarse_nodes.cpp:110]: remove else

File: src/define_coarse_nodes.cpp [hunk 3]
```diff
@@ -97,22 +97,15 @@ void level::define_coarse_nodes_onelevel(level* coarser)
...
-        for (int i = 0; i < ntheta_int; i += 2) {
-            coarse_nodes_list_r[j * ntheta_int + i]     = j;
-            coarse_nodes_list_theta[j * ntheta_int + i] = i;
-
-            indices_r.insert(j);
-            indices_theta.insert(i);
-        }
-    }
     for (int j = 0; j < nr_int + 1; j++) {
         for (int i = 0; i < ntheta_int; i++) {
```
Comment (mknaranja) [inline src/define_coarse_nodes.cpp:101]:
```suggestion
    for (int j = 0; j < nr_int + 1; j+=2) {
        for (int i = 0; i < ntheta_int; i+=2) {
```

File: src/define_coarse_nodes.cpp [hunk 4]
```diff
@@ -97,22 +97,15 @@ void level::define_coarse_nodes_onelevel(level* coarser)
...
-                    coarse_nodes_list_theta[j * ntheta_int + i] = -1;
-                }
+
+            if (i % 2 == 0 && j % 2 == 0) {
+                coarse_nodes_list_r[j * ntheta_int + i]     = j;
+                coarse_nodes_list_theta[j * ntheta_int + i] = i;
+
+                indices_r.insert(j);
+                indices_theta.insert(i);
             }
```
Comment (mknaranja) [inline src/define_coarse_nodes.cpp:109]:
```suggestion
         coarse_nodes_list_r[j * ntheta_int + i]     = j;
         coarse_nodes_list_theta[j * ntheta_int + i] = i;

         indices_r.insert(j);
         indices_theta.insert(i);
```

**Review by mknaranja — CHANGES_REQUESTED** [review]
Can you have a look at @CodingAllan ?

**Comment by CodingAllan** [comment]
Thanks for the review. The suggested changes look good. I updated it.

**Comment by mknaranja** [comment]
@CodingAllan There's a merge conflict, which is hopefully resolved when merging the new main. Could you try?

**Review by mknaranja — APPROVED** [review]

**Pull request merged**